datafusion/core/tests/sqllogictests/test_files/aggregate.slt

```diff
@@ -1710,13 +1854,25 @@ select sum(c1), arrow_typeof(sum(c1)) from d_table;
 ----
 100 Decimal128(20, 3)
 
+query TRT
+select c2, sum(c1), arrow_typeof(sum(c1)) from d_table GROUP BY c2 ORDER BY c2;
+----
+A 1100.045 Decimal128(20, 3)
```

Comment:
I'm not 100% sure whether the output type is correct, but I also don't know that it isn't so :+1:

Reply:
cc @liukun4515  in case you have some thoughts (this PR doesn't change the type FWIW)

Reply:
The data type of `c1` is `c1 decimal(10,3)`
the result of `sum(c1)` is `decimal(10+10,3)`, this align with spark rule: `the new precision = old precision + 10`